dask/array/fft.py

```diff
@@ -86,19 +70,73 @@ def _ihfft_out_chunks(a, n, axis):
     return chunks
 
 
-fft = _fft_wrap(npfft.fft, np.complex_, _fft_out_chunks)
+_out_chunk_fns = {'fft': _fft_out_chunks,
+                  'ifft': _fft_out_chunks,
+                  'rfft': _rfft_out_chunks,
+                  'irfft': _irfft_out_chunks,
+                  'hfft': _hfft_out_chunks,
+                  'ihfft': _ihfft_out_chunks}
 
 
-ifft = _fft_wrap(npfft.ifft, np.complex_, _fft_out_chunks)
+def fft_wrap(fft_func, kind=None, dtype=None):
+    """ Wrap 1D complex FFT functions
 
+    Takes a function that behaves like ``numpy.fft`` functions and
+    a specified kind to match it to that are named after the functions
+    in the ``numpy.fft`` API.
 
-rfft = _fft_wrap(npfft.rfft, np.complex_, _rfft_out_chunks)
+    Supported kinds include:
 
+        * fft
+        * ifft
+        * rfft
+        * irfft
+        * hfft
+        * ihfft
 
-irfft = _fft_wrap(npfft.irfft, np.float_, _irfft_out_chunks)
+    Examples
+    --------
+    >>> parallel_fft = fft_wrap(np.fft.fft)
+    >>> parallel_ifft = fft_wrap(np.fft.ifft)
+    """
+    if scipy is not None:
+        if fft_func is scipy.fftpack.rfft:
+            raise ValueError("SciPy's `rfft` doesn't match the NumPy API.")
+        elif fft_func is scipy.fftpack.irfft:
+            raise ValueError("SciPy's `irfft` doesn't match the NumPy API.")
+
+    if kind is None:
+        kind = fft_func.__name__
+    try:
+        out_chunk_fn = _out_chunk_fns[kind]
+    except KeyError:
+        raise ValueError("Given unknown `kind` %s." % kind)
+
+    _dtype = dtype
+
+    def func(a, n=None, axis=-1):
+        dtype = _dtype
+        if dtype is None:
+            dtype = fft_func(np.ones(8, dtype=a.dtype)).dtype
 
+        if len(a.chunks[axis]) != 1:
+            raise ValueError(chunk_error % (axis, a.chunks[axis]))
 
-hfft = _fft_wrap(npfft.hfft, np.float_, _hfft_out_chunks)
+        chunks = out_chunk_fn(a, n, axis)
+
+        return a.map_blocks(fft_func, n=n, axis=axis, dtype=dtype,
+                            chunks=chunks)
+
+    np_name = fft_func.__name__
+    if fft_func.__doc__ is not None:
+        func.__doc__ = (fft_preamble % (np_name, np_name)) + fft_func.__doc__
+    func.__name__ = np_name
```

Comment:
Missed fixing the docstring for these wrappers to note that these functions could from different locations. Change was made in PR ( https://github.com/dask/dask/pull/2110 ).